pkg/container-collection/networktracer/connect.go

```diff
@@ -16,16 +16,28 @@ package networktracer
 
 import (
 	"fmt"
-	"strconv"
 
 	"github.com/google/uuid"
 	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
 	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
 )
 
-type Tracer[Event any] interface {
-	Attach(key string, pid uint32, eventCallback func(Event)) error
-	Detach(key string) error
+type Tracer interface {
+	// Tracer must also implement either CallbackApproach or PopApproach
+	Detach(pid uint32) error
+}
+
+// CallbackApproach represents network tracers that provide the events through a
+// callback as soon as they are available.
+type CallbackApproach[Event any] interface {
+	Attach(pid uint32, eventCallback func(Event)) error
+}
+
+// PopApproach represents network tracers that don't return the events through a
+// callback but instead they provide a method to retrieve them.
+type PopApproach[Event any] interface {
+	Attach(pid uint32) error
+	Pop() ([]*Event, error)
 }
```

Comment:
With this implementation, the compiler doesn't fail if the tracer doesn't implement any of the `Attach` approaches, but it will fail at runtime. WDYT?

Reply:
Even though i dislike the idea of ditching compile time safety, i currently also don't know another way to support the 2 differenct `Tracer`s through one `interface` easily.